packages/app/components/document/form-actions.tsx

```diff
@@ -189,9 +198,24 @@ const FormActions = ({
                     <Button
                         className={styles.button}
                         variant="secondary"
-                        onClick={handleSave}
+                        onClick={async () => {
+                            setIsSaving(true)
```

Comment:
I should use a `setTimeout` here to remove the spinner in case the document fails to validate and save.